src/utils/converter.ts

```diff
@@ -1,20 +1,32 @@
-import { utils } from 'near-api-js';
-import { Gas } from 'near-units';
-import { B_div_B, B_mul_N } from './math'
-
-
-const { formatNearAmount, parseNearAmount } = utils.format;
-
-const toSmall = (big: number | string, decimals: number): bigint => B_mul_N(BigInt("1" + "0".repeat(decimals)), big);
-const toLarge = (small: bigint, decimals: number): number => B_div_B(small, BigInt("1" + "0".repeat(decimals)))
+import { Big } from 'big.js';
+
+
+// config for Big.js behavior. see: https://mikemcl.github.io/big.js/
+Big.RM = 0;
+Big.DP = 40;
+Big.NE = -40;
+Big.PE = 40;
+
+const removeTrailingZeros = (amount: string): string => amount.replace(/\.?0*$/, '');
+// token amount -> indivisible units
+const parseTokenAmount = (amount: number | string, decimals: number): string => Big(amount).times(Big(10).pow(decimals)).toFixed();
+// indivisible units -> token amount
+const formatTokenAmount = (amount: number | string, decimals: number, precision: number): string => {
+    const formattedAmount: string = Big(amount).div(Big(10).pow(decimals)).toFixed(precision);
+    return removeTrailingZeros(formattedAmount);
+}
 
-const toTGas = (gas: string | number): number => parseFloat(gas.toString()) * 1e-12; 
-const toGas = (TGas: string | number): number => parseInt( Gas.parse(`${TGas} TGas`).toString() );
+const toTGas = (amount: string | number): number => parseInt( formatTokenAmount(amount, 12, 12) ); 
```

Comment:
Did you mean `parseFloat`?

Reply:
Good catch! thank you.
We can lose precision when parsing JS floats, see big.js [docs example](https://mikemcl.github.io/big.js/#strict)
The better fix would be to use ArgsBig for Gas, and any other numbers where precision is critical.

Reply:
@lennczar changed Gas to ArgsBig, so the methods have string return type now